pkg/logs/logs_test.go

```diff
@@ -0,0 +1,453 @@
+		logger.Error(errors.New("fake-error"), "Contextual error", "key", "value")
+
+		klog.FlushAndExit(time.Second, 0)
+	}
+
+	tests := []struct {
+		name          string
+		flags         string
+		skipIntialize bool
+		expectError   bool
+		expectStdout  string
+		expectStderr  string
+	}{
+		{
+			name:        "help",
+			flags:       "-h",
+			expectError: true,
+			expectStdout: `
+pflag: help requested
+`,
+			expectStderr: `
+Usage of test-logs:
+  -v, --log-level Level         number for the log level verbosity
+      --logging-format string   Sets the log format. Permitted formats: "json", "text". (default "json")
+      --vmodule pattern=N,...   comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)
+`,
+		},
+		{
+			name:        "unrecognized-flag",
+			flags:       "--foo",
+			expectError: true,
+			expectStdout: `
+unknown flag: --foo
+`,
+			expectStderr: `
+unknown flag: --foo
+Usage of test-logs:
+  -v, --log-level Level         number for the log level verbosity
+      --logging-format string   Sets the log format. Permitted formats: "json", "text". (default "json")
+      --vmodule pattern=N,...   comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)
+`,
+		},
+		{
+			name:        "v-long-form-not-available",
+			flags:       "--v=3",
+			expectError: true,
+			expectStdout: `
+unknown flag: --v
+`,
+			expectStderr: `
+unknown flag: --v
+Usage of test-logs:
+  -v, --log-level Level         number for the log level verbosity
+      --logging-format string   Sets the log format. Permitted formats: "json", "text". (default "json")
+      --vmodule pattern=N,...   comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)
+`,
+		},
+		{
+			name:        "logging-format-unrecognized",
+			flags:       "--logging-format=foo",
+			expectError: true,
+			expectStderr: `
+Error in logging configuration: format: Invalid value: "foo": Unsupported log format
+`,
+		},
+		{
+			name:          "original-defaults",
+			flags:         "",
+			skipIntialize: true,
+			expectStderr: `
+vCert: 0000/00/00 00:00:00 log Print
+vCert: 0000/00/00 00:00:00 INFO slog Info
+vCert: 0000/00/00 00:00:00 WARN slog Warn
+vCert: 0000/00/00 00:00:00 ERROR slog Error
+I0000 00:00:00.000000   00000 logs_test.go:000] klog Info
+W0000 00:00:00.000000   00000 logs_test.go:000] klog Warning
+E0000 00:00:00.000000   00000 logs_test.go:000] "klog Error" err="fake-error"
+I0000 00:00:00.000000   00000 logs_test.go:000] "klog InfoS" key="value"
+E0000 00:00:00.000000   00000 logs_test.go:000] "Contextual error" err="fake-error" logger="foo" key="value"
+`,
+		},
+		{
+			name:  "modified-defaults",
+			flags: "",
+			expectStdout: `
+{"ts":0000000000000.000,"caller":"logs/logs.go:000","msg":"log Print","source":"vcert","v":0}
+{"ts":0000000000000.000,"caller":"logs/logs_test.go:000","msg":"slog Info","v":0}
```

Comment:
I wish we were able to show the real source of the log line rather than the generic `logs/logs_test.go`. I guess we can improve this once we have migrated the `logs.Log.Printf` calls to `slog.Info`.